SimTracker/TrackAssociation/plugins/TrackTimeValueMapProducer.cc

```diff
@@ -0,0 +1,270 @@
+// This producer assigns vertex times (with a specified resolution) to tracks.
+// The times are produced as valuemaps associated to tracks, so the track dataformat doesn't
+// need to be modified.
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/Common/interface/View.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrackFwd.h"
+#include "SimDataFormats/Track/interface/SimTrackContainer.h"
+#include "SimDataFormats/Vertex/interface/SimVertexContainer.h"
+#include "SimTracker/TrackerHitAssociation/interface/TrackerHitAssociator.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingVertex.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingVertexContainer.h"
+
+#include "SimDataFormats/Associations/interface/TrackToTrackingParticleAssociator.h"
+
+#include <memory>
+
+#include "ResolutionModel.h"
+#include "CLHEP/Units/SystemOfUnits.h"
+#include "FWCore/Utilities/interface/isFinite.h"
+#include "CLHEP/Random/RandGauss.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+
+#include "FWCore/Utilities/interface/isFinite.h"
+#include "FWCore/Utilities/interface/transform.h"
+
+class TrackTimeValueMapProducer : public edm::EDProducer {
```

Comment:
this should be of a type known to threaded framework
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes
edm::global would be a better choice if implementation allows, edm::stream is OK.
edm::one is rather an exception and will need some justification.